src/uct/ib/rdmacm/rdmacm_ep.c

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "rdmacm_ep.h"
+
+static UCS_CLASS_INIT_FUNC(uct_rdmacm_ep_t, uct_iface_t *tl_iface,
+                           const ucs_sock_addr_t *sockaddr,
+                           uct_sockaddr_conn_reply_callback_t reply_cb,
+                           void *arg, uint32_t cb_flags,
+                           const void *priv_data, size_t length)
+{
+    uct_rdmacm_iface_t *iface = ucs_derived_of(tl_iface, uct_rdmacm_iface_t);
+    size_t ip_len = ucs_max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN);
+    char *ip_str  = ucs_alloca(ip_len);
+    uct_rdmacm_priv_data_hdr_t hdr;
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super);
+
+    ucs_assert(iface->is_server == 0);
+
+    /* Initialize these fields before calling rdma_resolve_addr to avoid a race
+     * where they are used before being initialized (from the async thread
+     * - after an RDMA_CM_EVENT_ROUTE_RESOLVED event) */
+    hdr.length           = length;
+    self->conn_reply_cb  = reply_cb;
+    self->conn_reply_arg = arg;
+    self->priv_data      = ucs_malloc(sizeof(uct_rdmacm_priv_data_hdr_t) + length,
+                                      "client private data");
+
+    memcpy(self->priv_data, &hdr, sizeof(uct_rdmacm_priv_data_hdr_t));
+    memcpy(self->priv_data + sizeof(uct_rdmacm_priv_data_hdr_t), priv_data, length);
+
+    /* The interface can point at one endpoint at a time and therefore, the
+     * connection establishment cannot be done in parallel for several endpoints */
+    if (iface->ep == NULL) {
+        iface->ep = self;
+    } else {
+        /* Add the ep to the pending queue */
+        self->remote_addr = (struct sockaddr *)(sockaddr->addr);
```

Comment:
why do we store remote adress only for pending eps?

Reply:
in order to use it when extracting the ep from the pending queue. so that it would have the remote address to resolve the route to.
but it's only preparation for the next pr. in this pr it's not used. better if i remove it for now?